myapp/templates/admin_edit_data.html

```diff
@@ -0,0 +1,127 @@
+{% extends "admin_home.html" %}
+{% block head %}
+    <meta charset="UTF-8">
+    <title>Edit Data</title>
+<script>
+	$(document).ready(function(){
+		
+		$('button[name=save_all]').click(function(){
+
+			//Collect all the data in the table
+			old_data_ids_list = []
+			new_data_list = []
+			var all_data = document.getElementsByTagName('input');
+			for (var i=0;i<all_data.length;i++){
+				if ($(all_data[i]).attr('class')!='student_name'){
+					old_data_ids_list.push($(all_data[i]).data('dataid'));
+					var row_id = $(all_data[i]).data('rowid');
+					var student_name = $(all_data[i]).closest('tr').find(".student_name").val();
+					console.log(student_name);
+					var data_id = Generate_data_id(row_id,student_name);
+					new_data_list.push({'row_id':row_id,'data_id':data_id,'student_name':student_name,'row_data':$(all_data[i]).val()})
+
+				}
+			}		
+			
+			$.ajax({
+			  type: 'POST',
+			  contentType: 'application/json',
+			  dataType: 'json',
+			  url: 'http://127.0.0.1:5000/_admin_change_data',
```

Comment:
i believe you should be able to post to `/_admin_change_data`, which will proof your client code against changes in your deployment architecture.